File: API/Models/Book.cs
namespace API.Models;

public class Book
{
    public Guid Id { get; set; }
    public string? Olid { get; set; }
    public string Title { get; set; }
    public string Author { get; set; }
    public string? Description { get; set; }
    public string? CoverPath { get; set; }
    public ReadingStatus Status { get; set; }
    public int Progress { get; set; }
    public int TotalPages { get; set; }
    public DateTime? LastRead { get; set; }
}

File: src/lib/server/types/book.ts

export type Book = {
    id: string;
    title: string | null;
    author: string | null;
    description: string | null;
    read_status: "to-read" | "reading" | "finished" | "dropped";
    cover_path: string | null;
    progress: number | null;
    dir: string | null;
    cover?: string; // Computed field for UI
}